### PR TITLE
fix: specify compiler prop in pom

### PR DESCRIPTION
**Pull request by saranyailla**
**Issue #, if available:**

**Description of changes:**

**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

**Checklist:**
- [ ] Updated the README if applicable
- [ ] Updated or added new unit tests
- [ ] Updated or added new integration tests
- [ ] Updated or added new end-to-end tests
- [ ] If your code makes a remote network call, it was tested with a proxy
- [ ] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/python/HelloWorld/gdk-config.json [hunk 1]
```diff
@@ -4,7 +4,10 @@
       "author": "<PLACEHOLDER_AUTHOR>",
       "version": "NEXT_PATCH",
       "build": {
-        "build_system": "zip"
+        "build_system": "zip",
+        "options": {
+          "zip_name": "HelloWorld"
```
Comment (MikeDombo) [inline templates/python/HelloWorld/gdk-config.json:9]: leaving the name empty will use the component name right? That's what we want don't we?

Comment (saranyailla) [reply templates/python/HelloWorld/gdk-config.json:9]: Artifact name in the recipe must be updated to component name as well. I'm just trying to keep it as close as to what the customers on previous versions of GDK use. I've no strong pref.

Comment (MikeDombo) [reply templates/python/HelloWorld/gdk-config.json:9]: Artifact name in recipe should be replaced automatically using `COMPONENT_NAME` right?

Comment (MikeDombo) [reply templates/python/HelloWorld/gdk-config.json:9]: That would be my ideal.

Comment (saranyailla) [reply templates/python/HelloWorld/gdk-config.json:9]: GDK doesn't update the artifact names in the recipe (for any build system). It just updates the S3 URI path. It checks if an artifact with that name is present in the build files (`target`, `build/libs` or`zip-build`)

Comment (MikeDombo) [reply templates/python/HelloWorld/gdk-config.json:9]: I mean, a variable replacement. Don't we do a replacement for COMPONENT_NAME?

Comment (saranyailla) [reply templates/python/HelloWorld/gdk-config.json:9]: Not really. We update `componentName`property in the recipe.

File: templates/python/HelloWorld/gdk-config.json [hunk 2]
```diff
@@ -4,7 +4,10 @@
...
+        }
```
Comment (MikeDombo) [inline templates/python/HelloWorld/gdk-config.json:10]: update the minimum GDK version.

File: templates/python/LocalPubSub/gdk-config.json [hunk 3]
```diff
@@ -13,4 +16,4 @@
     }
   },
   "gdk_version": "1.0.0"
```
Comment (MikeDombo) [inline templates/python/LocalPubSub/gdk-config.json:18]: update GDK version.

**Review by MikeDombo — APPROVED** [review]

**Review by robcmann — APPROVED** [review]

**Pull request merged**